src/core/generator-setup.c

```diff
@@ -35,18 +35,21 @@ void lookup_paths_trim_generator(LookupPaths *p) {
                 (void) rmdir(p->generator_late);
 }
 
-void lookup_paths_flush_generator(LookupPaths *p) {
+void lookup_paths_flush_generator(LookupPaths *p, bool test) {
```

Comment:
hmm, another idea would be to store LookupPathsFlags inside of LookupPaths as a new field, and then check for some existing or new flag to control this?